### PR TITLE
Rendering: Set window dirty only on app thread

### PR DESCRIPTION
**Pull request by enen92**
## Description
Alternative to https://github.com/xbmc/xbmc/pull/25121 to avoid polling on the rendering thread, goes via app messaging instead.

### CODE REVIEW

File: xbmc/messaging/ApplicationMessenger.h [hunk 1]
```diff
@@ -90,6 +90,7 @@
 
 #define TMSG_SET_VOLUME                   TMSG_MASK_APPLICATION + 36
 #define TMSG_SET_MUTE                     TMSG_MASK_APPLICATION + 37
+#define TMSG_RESTART_RENDERING TMSG_MASK_APPLICATION + 38
```
Comment (sarbes) [inline xbmc/messaging/ApplicationMessenger.h:93]: Maybe adjust the indentation?

Comment (fritsch) [reply xbmc/messaging/ApplicationMessenger.h:93]: clang-format hook is happy.

Comment (enen92) [reply xbmc/messaging/ApplicationMessenger.h:93]: yeap, other lines also don't follow the identation. It's better for someone to do this later as well as killing the defines (out of scope of this PR though).

Comment (sarbes) [reply xbmc/messaging/ApplicationMessenger.h:93]: @fritsch clang formatting is turned off, so it shouldn't complain.

But yeah, it's just nitpicking.

**Comment by ksooo** [comment]
@enen92 @sarbes heads up: with this patch applied, Android (yes, Android, no typo) version of Kodi does not start anymore. It crashes.

```
05-05 18:47:05.538   566   566 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xb8 in tid 566 (org.xbmc.kodi), pid 566 (org.xbmc.kodi)
05-05 18:47:05.597   634   634 I chatty  : uid=10078(org.xbmc.kodi) expire 2 lines
05-05 18:47:05.598  3325  3325 I tombstoned: received crash request for pid 566
05-05 18:47:05.617   634   634 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
05-05 18:47:05.617   634   634 F DEBUG   : Build fingerprint: 'NVIDIA/mdarcy/mdarcy:11/RQ1A.210105.003/7825230_3167.5736:user/release-keys'
05-05 18:47:05.617   634   634 F DEBUG   : Revision: '0'
05-05 18:47:05.617   634   634 F DEBUG   : ABI: 'arm64'
05-05 18:47:05.618   634   634 F DEBUG   : Timestamp: 2024-05-05 18:47:05+0200
05-05 18:47:05.618   634   634 F DEBUG   : pid: 566, tid: 566, name: org.xbmc.kodi  >>> org.xbmc.kodi <<<
05-05 18:47:05.618   634   634 F DEBUG   : uid: 10078
05-05 18:47:05.618   634   634 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xb8
05-05 18:47:05.618   634   634 F DEBUG   : Cause: null pointer dereference
05-05 18:47:05.618   634   634 F DEBUG   :     x0  0000000000000000  x1  0000007ffff0e6e8  x2  0000000000000000  x3  0000000000000010
05-05 18:47:05.618   634   634 F DEBUG   :     x4  0000000000002a26  x5  0000002c48f76449  x6  3b65636166727553  x7  7f7f7f7f7f7f7f7f
05-05 18:47:05.618   634   634 F DEBUG   :     x8  dfa7191972fb9b3b  x9  dfa7191972fb9b3b  x10 0000000000000001  x11 0000000000000000
05-05 18:47:05.618   634   634 F DEBUG   :     x12 000000000000000a  x13 0000002cb65b520b  x14 0000002c46bbe4fc  x15 0000000000000007
05-05 18:47:05.618   634   634 F DEBUG   :     x16 0000002cb79c8170  x17 00000029b6845ce0  x18 00000029b0d1c000  x19 0000000000000000
05-05 18:47:05.618   634   634 F DEBUG   :     x20 0000000000000000  x21 0000000000000000  x22 0000007ffff0e6e8  x23 0000000000000000
05-05 18:47:05.618   634   634 F DEBUG   :     x24 00000029b3bb0d30  x25 00000029b140a000  x26 0000000000000000  x27 0000000000000002
05-05 18:47:05.618   634   634 F DEBUG   :     x28 0000000000000002  x29 0000007ffff0e6d0
05-05 18:47:05.618   634   634 F DEBUG   :     lr  0000002cb41fdfb4  sp  0000007ffff0e660  pc  0000002cb41fd2e4  pst 0000000060000000
05-05 18:47:05.864  3907  4370 I nv_bt_stats: btif_nv_stats_update_host_stats() addr: ff:ff:ff:ff:ff:ff , json string: cmd_timeout_rsp_after_to
05-05 18:47:06.012   634   634 F DEBUG   : backtrace:
05-05 18:47:06.012   634   634 F DEBUG   :       #00 pc 00000000012692e4  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/lib/arm64/libkodi.so!libkodi.so (offset 0xd09000) (CSlideShowPic::GetOriginalWidth()+8)
05-05 18:47:06.012   634   634 F DEBUG   :       #01 pc 0000000001269fb0  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/lib/arm64/libkodi.so!libkodi.so (offset 0xd09000) (CSlideShowPic::Process(unsigned int, std::__ndk1::vector<CDirtyRegion, std::__ndk1::allocator<CDirtyRegion> >&)+2504)
05-05 18:47:06.012   634   634 F DEBUG   :       #02 pc 0000000001269fb0  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/lib/arm64/libkodi.so!libkodi.so (offset 0xd09000) (CSlideShowPic::Process(unsigned int, std::__ndk1::vector<CDirtyRegion, std::__ndk1::allocator<CDirtyRegion> >&)+2504)
05-05 18:47:06.012   634   634 F DEBUG   :       #03 pc 000000000138c078  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/lib/arm64/libkodi.so!libkodi.so (offset 0xd09000) (CApplicationPlayer::FlushRenderer()+152)
05-05 18:47:06.012   634   634 F DEBUG   :       #04 pc 0000000000d94638  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/lib/arm64/libkodi.so!libkodi.so (offset 0xd09000) (CXBMCApp::onReceive(CJNIIntent)+964)
05-05 18:47:06.012   634   634 F DEBUG   :       #05 pc 00000000019c0868  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/lib/arm64/libkodi.so!libkodi.so (offset 0xd09000) (XFILE::CFileAndroidApp::Open(CURL const&)+120)
05-05 18:47:06.012   634   634 F DEBUG   :       #06 pc 00000000019c047c  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/lib/arm64/libkodi.so!libkodi.so (offset 0xd09000) (XFILE::CAndroidAppDirectory::GetDirectory(CURL const&, CFileItemList&)+1060)
05-05 18:47:06.012   634   634 F DEBUG   :       #07 pc 00000000000063a8  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/oat/arm64/base.odex (art_jni_trampoline+152)
05-05 18:47:06.012   634   634 F DEBUG   :       #08 pc 0000000000133564  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+548) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.012   634   634 F DEBUG   :       #09 pc 00000000001a8a78  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+200) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.012   634   634 F DEBUG   :       #10 pc 000000000031831c  /apex/com.android.art/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+376) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.012   634   634 F DEBUG   :       #11 pc 000000000030e648  /apex/com.android.art/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+996) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.012   634   634 F DEBUG   :       #12 pc 0000000000682f24  /apex/com.android.art/lib64/libart.so (MterpInvokeVirtualQuick+672) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.012   634   634 F DEBUG   :       #13 pc 0000000000131594  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_virtual_quick+20) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.012   634   634 F DEBUG   :       #14 pc 0000000000089fda  /data/app/~~nl7f-RGNfhgUptHg6o8Leg==/org.xbmc.kodi-pxiFtXBlRgUJ4KF62yHohg==/oat/arm64/base.vdex (org.xbmc.kodi.XBMCMainView.surfaceCreated+26)
05-05 18:47:06.012   634   634 F DEBUG   :       #15 pc 0000000000305c44  /apex/com.android.art/lib64/libart.so (art::interpreter::Execute(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool, bool) (.llvm.10194798506748459250)+268) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.012   634   634 F DEBUG   :       #16 pc 000000000066b1e8  /apex/com.android.art/lib64/libart.so (artQuickToInterpreterBridge+780) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.012   634   634 F DEBUG   :       #17 pc 000000000013cff8  /apex/com.android.art/lib64/libart.so (art_quick_to_interpreter_bridge+88) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #18 pc 00000000009f796c  /system/framework/arm64/boot-framework.oat (android.view.SurfaceView.updateSurface+4908) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #19 pc 00000000009f57e0  /system/framework/arm64/boot-framework.oat (android.view.SurfaceView.lambda$new$0$SurfaceView+96) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #20 pc 0000000000908a0c  /system/framework/arm64/boot-framework.oat (android.view.-$$Lambda$SurfaceView$w68OV7dB_zKVNsA-r0IrAUtyWas.onPreDraw+60) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #21 pc 0000000000842bf8  /system/framework/arm64/boot-framework.oat (android.view.ViewTreeObserver.dispatchOnPreDraw+264) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #22 pc 000000000096b4a4  /system/framework/arm64/boot-framework.oat (android.view.ViewRootImpl.performTraversals+11364) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #23 pc 000000000096edbc  /system/framework/arm64/boot-framework.oat (android.view.ViewRootImpl.doTraversal+156) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #24 pc 00000000008405ac  /system/framework/arm64/boot-framework.oat (android.view.ViewRootImpl$TraversalRunnable.run+60) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #25 pc 00000000008206a8  /system/framework/arm64/boot-framework.oat (android.view.Choreographer.doCallbacks+744) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #26 pc 0000000000820dc0  /system/framework/arm64/boot-framework.oat (android.view.Choreographer.doFrame+1232) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #27 pc 0000000000909af8  /system/framework/arm64/boot-framework.oat (android.view.Choreographer$FrameDisplayEventReceiver.run+72) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #28 pc 0000000000656bbc  /system/framework/arm64/boot-framework.oat (android.os.Handler.dispatchMessage+76) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #29 pc 000000000065a1fc  /system/framework/arm64/boot-framework.oat (android.os.Looper.loop+1772) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #30 pc 000000000041492c  /system/framework/arm64/boot-framework.oat (android.app.ActivityThread.main+748) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #31 pc 00000000001337e8  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+568) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #32 pc 00000000001a8a94  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+228) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #33 pc 00000000005556e4  /apex/com.android.art/lib64/libart.so (art::InvokeMethod(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned long)+1364) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #34 pc 00000000004d4ef0  /apex/com.android.art/lib64/libart.so (art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*)+52) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #35 pc 000000000008a6f4  /apex/com.android.art/javalib/arm64/boot.oat (art_jni_trampoline+180) (BuildId: fd3cf054c28fb5d3737a5d92c072cfb27c32c982)
05-05 18:47:06.013   634   634 F DEBUG   :       #36 pc 00000000008b1d78  /system/framework/arm64/boot-framework.oat (com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+136) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #37 pc 00000000008ba8b4  /system/framework/arm64/boot-framework.oat (com.android.internal.os.ZygoteInit.main+2212) (BuildId: 52d49727bd26336e38f46c7d1f160f42ffb4d1dc)
05-05 18:47:06.013   634   634 F DEBUG   :       #38 pc 00000000001337e8  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+568) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #39 pc 00000000001a8a94  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+228) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #40 pc 0000000000554120  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeWithVarArgs<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, std::__va_list)+448) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #41 pc 00000000005545d4  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeWithVarArgs<_jmethodID*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+92) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #42 pc 0000000000438b08  /apex/com.android.art/lib64/libart.so (art::JNI<true>::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+656) (BuildId: a0c6c489c777bbf271c42844824565bb)
05-05 18:47:06.013   634   634 F DEBUG   :       #43 pc 0000000000099424  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+124) (BuildId: b5a27241672fc9fb24f73218138b0359)
05-05 18:47:06.013   634   634 F DEBUG   :       #44 pc 00000000000a08e4  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+836) (BuildId: b5a27241672fc9fb24f73218138b0359)
05-05 18:47:06.013   634   634 F DEBUG   :       #45 pc 0000000000003580  /system/bin/app_process64 (main+1336) (BuildId: bdcc463a75a1be87eb0f7478ec16547d)
05-05 18:47:06.013   634   634 F DEBUG   :       #46 pc 000000000004988c  /apex/com.android.runtime/lib64/bionic/libc.so (__libc_init+108) (BuildId: 45994aaacbd3639b43f2bde9e3052edc)
```

**Comment by sarbes** [comment]
I guess we should check if the GUI is up, similar to: 
https://github.com/xbmc/xbmc/blob/d74b274138954c2d28f2e2a40aa298c1951ea261/xbmc/application/Application.cpp#L1519

If it is not up, the elements should be dirty by default when starting.

**Comment by enen92** [comment]
Honestly I wonder if we need this call at all (not needed on macos) and I also doubt it's needed on other platforms (added the workaround not to change the main logic).
From what I could understand by looking at the history this was introduced because of android long ago but it works just fine without this

**Comment by enen92** [comment]
Having said that, what do you guys think about removing this and let it settle for a few days to see if any regression occurs?
This was added in https://github.com/xbmc/xbmc/commit/bd4c0ae108ac4e2e4840c51f6ad1efd5699f89db as part of https://github.com/xbmc/xbmc/pull/1493

Looking at the diff it seems it was added because of rpi/omxplayer which is no longer part of Kodi. Feedback would be appreciated.

**Comment by sarbes** [comment]
Removing the MarkDirty() call entirely? Sure.

**Comment by enen92** [comment]
> Removing the MarkDirty() call entirely? Sure.

Exactly. I don't think it is necessary anymore (only feedback will tell) but even if it is I don't think it is done on the proper place and the eventual bug should be fixed properly. Mmarking dirty should be the reaction to some event (e.g. GUI available/app focused, etc) and not as a result of flagging render on/off from common code. Currently Kodi at the end of a resize operation already marks the window broken.

**Comment by enen92** [comment]
Superseeded by https://github.com/xbmc/xbmc/pull/25164

**Pull request closed without merging**